plugins/convert/mask/mask_blend.py

```diff
@@ -15,38 +14,24 @@ def __init__(self, mask_type, output_size, predicted_available, **kwargs):
         self.do_erode = self.config.get("erosion", 0) != 0
         self.do_blend = self.config.get("type", None) is not None
 
-    def process(self, detected_face, predicted_mask=None):
+    def process(self, new_face):
         """ Return mask and perform processing """
-        mask = self.get_mask(detected_face, predicted_mask)
-        raw_mask = mask.copy()
-        if not self.skip and self.do_erode:
-            mask = self.erode(mask)
-        if not self.skip and self.do_blend:
-            mask = self.blend(mask)
-        raw_mask = np.expand_dims(raw_mask, axis=-1) if raw_mask.ndim != 3 else raw_mask
-        mask = np.expand_dims(mask, axis=-1) if mask.ndim != 3 else mask
-        logger.trace("mask shape: %s, raw_mask shape: %s", mask.shape, raw_mask.shape)
-        return mask, raw_mask
-
-    def get_mask(self, detected_face, predicted_mask):
-        """ Return the mask from lib/model/masks and intersect with box """
-        if self.mask_type == "none":
-            # Return a dummy mask if not using a mask
-            mask = np.ones_like(self.dummy[:, :, 1])
-        elif self.mask_type == "predicted":
-            mask = predicted_mask
-        else:
-            landmarks = detected_face.reference_landmarks
-            mask = getattr(model_masks, self.mask_type)(landmarks, self.dummy, channels=1).mask
-        np.nan_to_num(mask, copy=False)
-        np.clip(mask, 0.0, 1.0, out=mask)
+        mask = new_face[..., 3:4]
+        if not self.skip:
+            if self.do_erode:
+                mask = self.erode(mask)
+            if self.do_blend:
+                mask = self.blend(mask)
+        mask = mask[..., None] if mask.ndim != 3 else mask
+        np.clip(mask, 0., 1., out=mask)
+        logger.trace("mask shape: %s", mask.shape)
         return mask
```

Comment:
Thread passes a 4 channel image into convert. Get_mask is not needed. Simplify mask splitting